### PR TITLE
boards: arm: stm32l562e_dk_ns: Add flash partitions

### PR DESCRIPTION
**Pull request by nandojve**
The current stm32l562_dk_ns has no flash partitions defined. This add missing flash partitions at board level following partition sizes that are compatible with the TF-M platform defined at flash_layout.h and removes the redundant overlays files.

### CODE REVIEW

File: boards/arm/stm32l562e_dk/stm32l562e_dk_ns.dts [hunk 1]
```diff
@@ -18,11 +18,35 @@
...
+		};
+		/* Secure image primary slot */
+		slot0_partition: partition@19000 {
+			label = "image-0";
+			reg = <0x00019000 0x0003C000>;
+		};
+		/* Non-secure image primary slot */
+		slot1_partition: partition@55000 {
+			label = "image-1";
+			reg = <0x00055000 0x0002A000>;
```
Comment (FRASTM) [inline boards/arm/stm32l562e_dk/stm32l562e_dk_ns.dts:49]: 0x2A000 could be written with DT_SIZE_K(x) format

File: boards/arm/stm32l562e_dk/stm32l562e_dk_ns.dts [hunk 2]
```diff
@@ -18,11 +18,35 @@
...
+
+		boot_partition: partition@0 {
+			label = "mcuboot";
+			reg = <0x00000000 0x00019000>;
+			read-only;
+		};
+		/* Secure image primary slot */
+		slot0_partition: partition@19000 {
+			label = "image-0";
+			reg = <0x00019000 0x0003C000>;
```
Comment (FRASTM) [inline boards/arm/stm32l562e_dk/stm32l562e_dk_ns.dts:44]: 0x3C000 could be written with DT_SIZE_K(x) format

**Review by erwango — COMMENTED** [review]
LGTM but following https://github.com/zephyrproject-rtos/zephyr/pull/49967, I wonder if partition should rather be called `slot1_ns_partition`.
Otherwise, splitting overlays in sample was a decision of the previous TF-M maintainer, for some reasons of his that I can't recall.
So I'd like current maintainer (@microbuilder) to confirm this is the good direction.

**Review by FRASTM — COMMENTED** [review]
else LGTM

**Comment by microbuilder** [comment]
> LGTM but following #49967, I wonder if partition should rather be called `slot1_ns_partition`. Otherwise, splitting overlays in sample was a decision of the previous TF-M maintainer, for some reasons of his that I can't recall. So I'd like current maintainer (@microbuilder) to confirm this is the good direction.

I think consistent naming with `slot1_ns_partition` makes sense to me, certainly in the same device family.

I don't recall the thinking myself for splitting overlays. I don't really have strong feelings either way about it myself, so I'm OK with the current status quo unless you feel there's a good reason to change?

File: boards/arm/stm32l562e_dk/stm32l562e_dk_ns.dts (hunk 1 above)
Comment (nandojve) [reply boards/arm/stm32l562e_dk/stm32l562e_dk_ns.dts]: Sure, added as request. I introduced the storage partition.

**Comment by nandojve** [comment]
> > LGTM but following #49967, I wonder if partition should rather be called `slot1_ns_partition`. Otherwise, splitting overlays in sample was a decision of the previous TF-M maintainer, for some reasons of his that I can't recall. So I'd like current maintainer (@microbuilder) to confirm this is the good direction.
> 
> I think consistent naming with `slot1_ns_partition` makes sense to me, certainly in the same device family.
> 
> I don't recall the thinking myself for splitting overlays. I don't really have strong feelings either way about it myself, so I'm OK with the current status quo unless you feel there's a good reason to change?

Renamed to slot1_ns_partition as requested and `label = "image_1_nonsecure";`

**Comment by microbuilder** [comment]
> Renamed to slot1_ns_partition as requested and `label = "image_1_nonsecure";`

Is the `label` property strictly required here? There's a conscious attempt to move away from label **properties** (`image_1_nonsecure` here) in place of **node labels** (`slot1_ns_partition`), so there should be a specific justification to keep them here.

This is true of other overlays, which should be cleaned up as well, but perhaps better to start a better precedent here?

**Review by microbuilder — DISMISSED** [review]

**Comment by nandojve** [comment]
> > Renamed to slot1_ns_partition as requested and `label = "image_1_nonsecure";`
> 
> Is the `label` property strictly required here? There's a conscious attempt to move away from label **properties** (`image_1_nonsecure` here) in place of **node labels** (`slot1_ns_partition`), so there should be a specific justification to keep them here.
> 
> This is true of other overlays, which should be cleaned up as well, but perhaps better to start a better precedent here?
> 
> EDIT: I won't block on this, though, given that a general cleanup is required for this on other boards.

I'm not sure about it but my feeling is that test still rely in the label. Now I'm doubt about `slot1_ns_partition` vs `image_1_nonsecure`. It is not clear what will happen when label is dropped for partitions.

https://github.com/zephyrproject-rtos/zephyr/blob/78d8ea4a6244279e3e866ec97543815d69f512e8/tests/drivers/flash/testcase.yaml#L29-L35

**Comment by erwango** [comment]
> I'm not sure about it but my feeling is that test still rely in the label. Now I'm doubt about `slot1_ns_partition` vs `image_1_nonsecure`. It is not clear what will happen when label is dropped for partitions.

Actually for STM32, the filter is:
https://github.com/zephyrproject-rtos/zephyr/blob/78d8ea4a6244279e3e866ec97543815d69f512e8/tests/drivers/flash/testcase.yaml#L41-L58

I agree this part deserves a cleanup.
But the naming itself comes from https://github.com/zephyrproject-rtos/zephyr/pull/48174 and I think it should be more widely adopted.

**Comment by nandojve** [comment]
@erwango ,

I dropped the labels, it seems not necessary anymore. Let me know if you want add this stm32l562e_dk or stm32l562e_dk_ns in the integration_platforms list.

**Comment by erwango** [comment]
> I dropped the labels, it seems not necessary anymore. Let me know if you want add this stm32l562e_dk or stm32l562e_dk_ns in the integration_platforms list.

Would be nice to add both indeed. Thanks

**Comment by nandojve** [comment]
> > I dropped the labels, it seems not necessary anymore. Let me know if you want add this stm32l562e_dk or stm32l562e_dk_ns in the integration_platforms list.
> 
> Would be nice to add both indeed. Thanks

Done!

**Review by microbuilder — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Pull request merged**